terraform/README.md

```diff
@@ -62,3 +74,47 @@ In the environment-specific modules:
 - `main.tf` calls the shared Terraform code, but this is also a place where you can add any other services, resources, etc, which you would like to set up for that environment
 - `variables.tf` lists the variables that will be needed, either to pass through to the child module or for use in this module
 - `secrets.auto.tfvars` is a file which contains the information about the service-key and other secrets that should not be shared
+
+In the bootstrap module:
+- `providers.tf` lists the required providers
+- `main.tf` sets up s3 bucket to be shared across all environments. It lives in ct-prod to communicate that it should not be deleted
+- `variables.tf` lists the variables that will be needed. Most values are hard-coded in this module
+- `run.sh` Helper script to set up a space deployer and run terraform. The terraform action (`show`/`plan`/`apply`/`destroy`) is passed as an argument
+- `teardown_creds.sh` Helper script to remove the space deployer setup as part of `run.sh`
+
+## Bootstrap Process
+
+Bootstrap is used to create an s3 bucket for later terraform runs to store their state in.
+
+### To run the bootstrap the first time or make changes
+
+1. Run `./run.sh plan` to verify that the changes are what you expect
+1. Run `./run.sh apply` to create the shared config bucket and retrieve the bucket credentials
+1. Follow instructions under `Use bootstrap credentials`
+1. Run `./teardown_creds.sh` to remove the space deployer account used to create the s3 bucket
+
+### Retrieving existing bucket credentials
```

Comment:
I am very curious if this will work on other people's machines, or only on mine since I have the local state file.

Reply:
```
Service key space-deployer-key already exists
./run.sh: line 16: jq: command not found
./run.sh: line 17: jq: command not found
No state.
```

Looks like I don't have that json processor installed, although we could add that to the instructions

Reply:
This also does not work if you don't have the state file already. Taking a conversation about the best way to address that here: https://gsa-tts.slack.com/archives/C01TT2YNX0R/p1630505518005900